enterprise/frontend/src/metabase-enterprise/transforms/pages/TransformListPage/TransformListPage.tsx

```diff
@@ -1,90 +1,206 @@
+
+import S from "./TransformListPage.module.css";
+import { type TreeNode, getCollectionNodeId } from "./types";
+import { buildTreeData, getDefaultExpandedIds } from "./utils";
+
+function useTargetCollectionId(): number | null {
+  const location = useLocation();
+  return useMemo(() => {
+    const searchParams = new URLSearchParams(location.search ?? "");
+    const param = searchParams.get("collectionId");
+    return param ? parseInt(param, 10) : null;
+  }, [location.search]);
+}
 
 export const TransformListPage = () => {
   const dispatch = useDispatch();
+  const targetCollectionId = useTargetCollectionId();
+  const hasScrolledRef = useRef(false);
   const [searchQuery, setSearchQuery] = useState("");
-  const [debouncedSearchQuery] = useDebouncedValue(searchQuery, 300);
 
-  const { data: transforms, error, isLoading } = useListTransformsQuery({});
+  const { data: targetCollection } = useGetCollectionQuery(
+    targetCollectionId
+      ? { id: targetCollectionId, namespace: "transforms" }
+      : skipToken,
+  );
 
-  const filteredTransforms = useMemo(() => {
-    if (!transforms) {
-      return [];
-    }
+  const {
+    data: collections,
+    error: collectionsError,
+    isLoading: isLoadingCollections,
+  } = useListCollectionsTreeQuery({
+    namespace: "transforms",
+    "exclude-archived": true,
+  });
 
-    if (!debouncedSearchQuery) {
-      return transforms;
-    }
+  const {
+    data: transforms,
+    error: transformsError,
+    isLoading: isLoadingTransforms,
+  } = useListTransformsQuery({});
 
-    const query = debouncedSearchQuery.toLowerCase();
-    return transforms.filter(
-      (transform) =>
-        transform.name.toLowerCase().includes(query) ||
-        transform.target.name.toLowerCase().includes(query),
-    );
-  }, [transforms, debouncedSearchQuery]);
+  const isLoading = isLoadingCollections || isLoadingTransforms;
+  const error = collectionsError ?? transformsError;
 
-  const transformColumnDef = useMemo<ColumnDef<Transform, ReactNode>[]>(
+  const treeData = useMemo(
+    () => buildTreeData(collections, transforms),
+    [collections, transforms],
+  );
+
+  const defaultExpanded = useMemo(
+    () => getDefaultExpandedIds(targetCollectionId, targetCollection),
+    [targetCollectionId, targetCollection],
+  );
+
+  const columnDefs = useMemo<TreeTableColumnDef<TreeNode>[]>(
     () => [
       {
+        id: "name",
         accessorKey: "name",
-        meta: {
-          width: "auto",
-        },
         header: t`Name`,
-        cell: ({ getValue }) => (
-          <Group data-testid="transform-name" gap="sm">
-            <Icon name="transform" c="brand" />
-            {getValue()}
-          </Group>
+        minWidth: 320,
+        enableSorting: true,
+        cell: ({ row }) => (
+          <EntityNameCell
+            data-testid="tree-node-name"
+            icon={row.original.nodeType === "folder" ? "folder" : "transform"}
```

Comment:
Not sure, but usually we have a special function to resolve collection icon, as some collections can be verified and/or have custom icons.